File: PSI-2020-backend/PSI-2020-backend/Models/Database/Course.cs
using PSI_2020_backend.Models.Database.Bases;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PSI_2020_backend.Models.Database
{
    public class Course : ModelBase
    {
        public string Name { get; set; }
        public string EngName { get; set; }
        public int Semester { get; set; }
        public int SumHoursForLectures { get; set; }
        public int SumHoursForLaboratories { get; set; }
        public int SumHoursForProjects { get; set; }
        public int SumHoursForSeminaries { get; set; }
        public int SumHoursForExercises { get; set; }
        public SubjectCard SubjectCard { get; set; }
    }
}

File: frontend/src/app/models/course.ts
import { SubjectCard } from "./subject-card";

export class Course {
    name: string;
    engName: string;
    semester: number = 1;
    sumHoursForLectures: number;
    sumHoursForLaboratories: number;
    sumHoursForProjects: number;
    sumHoursForSeminaries: number;
    sumHoursForExercises: number;
    subjectCard?: SubjectCard;

    constructor(init?: Partial<Course>) {
        Object.assign(this, init);
    }
}
